From 83f825a3e7ed9436bdf293a2d83c692c08e10935 Mon Sep 17 00:00:00 2001
From: mistrmochov <matejdvoracek@fastmail.com>
Date: Tue, 5 Aug 2025 16:37:23 +0200
Subject: [PATCH 16/21] fixing gitignore again

---
 .gitignore | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.gitignore b/.gitignore
index e508d9d3c..4cadb490f 100644
--- a/.gitignore
+++ b/.gitignore
@@ -6,7 +6,7 @@ out
 *.log
 /config.prop
 /notes.md
-/native/src/core/.vscode
+native/src/core/.vscode
 
 # Built binaries
 native/out
-- 
2.52.0

