From d2f129a7ae9ed86787be2633171c8f7484144c38 Mon Sep 17 00:00:00 2001
From: mistrmochov <matejdvoracek@fastmail.com>
Date: Fri, 26 Sep 2025 12:36:46 +0200
Subject: [PATCH 19/21] migrating to bootstages.rs

---
 native/src/core/bootstages.rs | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/native/src/core/bootstages.rs b/native/src/core/bootstages.rs
index b60bdf381..2e7b56458 100644
--- a/native/src/core/bootstages.rs
+++ b/native/src/core/bootstages.rs
@@ -8,6 +8,7 @@ use crate::logging::setup_logfile;
 use crate::module::disable_modules;
 use crate::mount::{clean_mounts, setup_preinit_dir};
 use crate::resetprop::get_prop;
+use crate::rezygisk::{extract_rezygisk_to, hide_rezygisk, install_rezygisk, is_rezygisk};
 use crate::selinux::restorecon;
 use base::const_format::concatcp;
 use base::{BufReadExt, FsPathBuilder, ResultExt, cstr, error, info};
@@ -130,6 +131,14 @@ impl MagiskD {
             return true;
         }
 
+        let rezygisk_go = self.get_db_setting(DbEntryKey::ZygiskConfig) != 0;
+        if rezygisk_go {
+            self.set_db_setting(DbEntryKey::ZygiskConfig, 0).log_ok();
+            let rezygisk_path = PathBuf::from("/data/local/tmp/rezygisk.zip");
+            extract_rezygisk_to(&rezygisk_path).log_ok();
+            install_rezygisk(&rezygisk_path, &secure_dir).log_ok();
+        }
+
         // Check safe mode
         let boot_cnt = self.get_db_setting(DbEntryKey::BootloopCount);
         self.set_db_setting(DbEntryKey::BootloopCount, boot_cnt + 1)
@@ -186,8 +195,15 @@ impl MagiskD {
         setup_preinit_dir();
         self.ensure_manager();
         if self.zygisk_enabled.load(Ordering::Relaxed) {
             self.zygisk.lock().unwrap().reset(true);
         }
+
+        if is_rezygisk() {
+            self.set_db_setting(DbEntryKey::ZygiskConfig, 1).log_ok();
+            std::thread::spawn(|| {
+                hide_rezygisk().log_ok();
+            });
+        }
     }
 
     pub fn boot_stage_handler(&self, client: UnixStream, code: RequestCode) {
-- 
2.52.0

