From 6c900e9787c4ab94874a7448f2c806017208ed81 Mon Sep 17 00:00:00 2001
From: mistrmochov <matejdvoracek@fastmail.com>
Date: Fri, 02 Jan 2026 21:06:50 +0800
Subject: [PATCH 21/21] false busybox error

---
 native/src/core/bootstages.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/native/src/core/bootstages.rs b/native/src/core/bootstages.rs
index 5e4c8eca8..cf50bde2f 100644
--- a/native/src/core/bootstages.rs
+++ b/native/src/core/bootstages.rs
@@ -78,7 +78,7 @@
         tmp_bb.mkdirs(0o755).ok();
         tmp_bb.append_path("busybox");
         busybox.copy_to(tmp_bb).ok();
-        tmp_bb.follow_link().chmod(0o755).log_ok();
+        tmp_bb.follow_link().chmod(0o755).ok();
 
         // Install busybox applets
         Command::new(&tmp_bb)
